src/H4LTools.cc

```diff
@@ -0,0 +1,855 @@
+    
+    return bestMuonindex;
+}
+std::vector<unsigned int> H4LTools::goodElectrons2015_noIso_noBdt(std::vector<unsigned int> Electronindex){
+    std::vector<unsigned int> bestElectronindex;
+    for (unsigned int i=0; i<Electronindex.size(); i++){
+        if ((Electron_pt[Electronindex[i]])>elePtcut){
+            if(Electron_sip3d[Electronindex[i]]<sip3dCut){
+                if((fabs(Electron_dxy[Electronindex[i]])<0.5)&&(fabs(Electron_dz[Electronindex[i]])<1)){
+                    bestElectronindex.push_back(Electronindex[i]);
+                }
+            }
+        }
+    }
+
+    return bestElectronindex;
+}
+std::vector<bool> H4LTools::passTight_BDT_Id(){
+    std::vector<bool> tightid;
+    float cutVal,mvaVal;
+    cutVal = 1000;
+    mvaVal = -1;
+    //unsigned nE = (*nElectron).Get()[0];
+    for (unsigned int i=0; i<Electron_pt.size(); i++){
+        if(Electron_pt[i]<10){
+            if(fabs(Electron_eta[i])<0.8) cutVal = 0.9044286167;
+            if((fabs(Electron_eta[i])>=0.8)&&(fabs(Electron_eta[i])<1.479)) cutVal = 0.9094166886;
+            if(fabs(Electron_eta[i])>=1.479) cutVal = 0.9443653660;
+        }
+        else{
+            if(fabs(Electron_eta[i])<0.8) cutVal = 0.1968600840;
+            if((fabs(Electron_eta[i])>=0.8)&&(fabs(Electron_eta[i])<1.479)) cutVal = 0.0759172100;
+            if(fabs(Electron_eta[i])>=1.479) cutVal = -0.5169136775;
+        }
+
+        mvaVal = Electron_mvaFall17V2Iso_WP90[i];
+        if(mvaVal > cutVal){
+            tightid.push_back(true);
+        }
+        else{
+            tightid.push_back(false);
+        }
+    
+    }
+    
+    return tightid;
+    
+}
+std::vector<bool> H4LTools::passTight_Id(){
+    std::vector<bool> tightid;
+    //unsigned nMu = (*nMuon).Get()[0];
+    for (unsigned int i=0; i<Muon_Pt_Corrected.size(); i++){
+        if (Muon_Pt_Corrected[i]<200){
+            tightid.push_back(Muon_isPFcand[i]);
+        }
+        else{
+            tightid.push_back(Muon_isPFcand[i]||(((Muon_ptErr[i]/Muon_Pt_Corrected[i])<0.3)&&(fabs(Muon_dxy[i])<0.2)&&(fabs(Muon_dz[i])<0.5)&&(Muon_nTrackerLayers[i]>5)));
+        }
+
+    }
+
+    return tightid;
+}
+
+std::vector<unsigned int> H4LTools::goodFsrPhotons(){
+    std::vector<unsigned int> goodFsrPhoton;
+    //unsigned nFsr = (*nFsrPhoton).Get()[0];
+    for (unsigned int i=0; i<FsrPhoton_pt.size(); i++){
+        if((FsrPhoton_pt[i]>2)&&(fabs(FsrPhoton_eta[i])<2.4)&&(FsrPhoton_relIso03[i]<0.8)){
+            goodFsrPhoton.push_back(i);
+        }
+    }
+    return goodFsrPhoton;
+}
+
+std::vector<unsigned int> H4LTools::SelectedJets(std::vector<unsigned int> ele, std::vector<unsigned int> mu){
+    std::vector<unsigned int> goodJets;
+    //unsigned nJ = (*nJet).Get()[0];
+    for(unsigned int i=0;i<Jet_pt.size();i++){
+        int overlaptag=0;
+        TLorentzVector jettest;
+        jettest.SetPtEtaPhiM(Jet_pt[i],Jet_eta[i],Jet_phi[i],Jet_mass[i]);
+        for(unsigned int ie=0;ie<ele.size();ie++){
```

Comment:
@YujiLee301: You should clean with only those jets that passed your pre-selection. I mean the pT and eta cut should be applied before checking the deltaR with leptons.